src/Sentry/Transaction.cs

```diff
@@ -254,8 +257,13 @@ public void Finish()
         {
             EndTimestamp = DateTimeOffset.UtcNow;
 
+            if (Exception is {} exception)
```

Comment:
This means that if the exception is logged before we finish the transaction, it won't be bound.

Ideally we'd be able to "connect the exception instance to a span" and if CaptureException happens right after that, it would be connected.

```
void method(ISpan tran) {
var child = tran.StartChild(..);
try {
throw null;
}
catch (e) {
child.Finish(e);
// Will know in which exact Span that the exception happened
SentrySdk.CaptureException(e);
// don't rethrow
}
`

Reply:
This is how it works in Java btw.
When should it happen then? When `transaciton.Exception` changes?